web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java

```diff
@@ -1388,7 +1401,12 @@ public void run() {
 							if (isCurrentDatabase(DATABASE_MYSQL)) {
 								sql = "create database if not exists `?` default character set utf8";
 							} else if (isCurrentDatabase(DATABASE_POSTGRESQL)) {
-								sql = "create database `?` encoding 'utf8'";
+								if (databaseExistsInPostgres(wizardModel.databaseName, wizardModel.createDatabaseUsername, 
```

Comment:
If the same database exists, we should just fail like we do for mysql.

Reply:
In my most recent commit i fixed it. see: https://github.com/mherman22/openmrs-core/blob/a24f70af9eed835fa340681c300261bfada7ff77/web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java#L1404-L1407

Reply:
To simplify this work, i want us to start with the exact same functionality that we have in mysql. After which you will be free to suggest improvements. But then they will need to be implemented in both mysql and postgresql

Reply:
Makes sense, let me remove that!

Reply:
just one thing to put out there is that mysql behaves differently from postgres. mysql allows something like `"create database if not exists `?` default character set utf8";` unlike postgres.

So in this case, the same behaviour with mysql would mean keeping this `databaseExistsInPostgres` check. 

They only share similar behaviour for an existing user.